Traceback (most recent call last):
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danypineault/anaconda3/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/danypineault/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Retuns the total carbon emissions; Eq. 18
@njit('float64(float64[:],int32)') 
def fE(iEIND,index):
    return iEIND[index] + etree[index]

#Eq.14: Determines the emission of carbon by industry EIND
@njit('float64(float64[:],float64[:],float64[:],int32)') 
def fEIND(iYGROSS, iMIU, isigma,index):
    return isigma[index] * iYGROSS[index] * (1 - iMIU[index])

#Cumulative industrial emission of carbon
@njit('float64(float64[:],float64[:],int32)') 
def fCCA(iCCA,iEIND,index):
    return iCCA[index-1] + iEIND[index-1] * 5 / 3.666

#Cumulative total carbon emission
@njit('float64(float64[:],float64[:],int32)')
def fCCATOT(iCCA,icumetree,index):
    return iCCA[index] + icumetree[index]
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mTypingError[0m                               Traceback (most recent call last)
Cell [0;32mIn[8], line 2[0m
[1;32m      1[0m [38;5;66;03m# Retuns the total carbon emissions; Eq. 18[39;00m
[0;32m----> 2[0m [38;5;129m@njit[39m([38;5;124m'[39m[38;5;124mfloat64(float64[:],int32)[39m[38;5;124m'[39m) 
[1;32m      3[0m [38;5;28;01mdef[39;00m [38;5;21mfE[39m(iEIND,index):
[1;32m      4[0m     [38;5;28;01mreturn[39;00m iEIND[index] [38;5;241m+[39m etree[index]
[1;32m      6[0m [38;5;66;03m#Eq.14: Determines the emission of carbon by industry EIND[39;00m

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/decorators.py:241[0m, in [0;36m_jit.<locals>.wrapper[0;34m(func)[0m
[1;32m    239[0m     [38;5;28;01mwith[39;00m typeinfer[38;5;241m.[39mregister_dispatcher(disp):
[1;32m    240[0m         [38;5;28;01mfor[39;00m sig [38;5;129;01min[39;00m sigs:
[0;32m--> 241[0m             disp[38;5;241m.[39mcompile(sig)
[1;32m    242[0m         disp[38;5;241m.[39mdisable_compile()
[1;32m    243[0m [38;5;28;01mreturn[39;00m disp

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/dispatcher.py:965[0m, in [0;36mDispatcher.compile[0;34m(self, sig)[0m
[1;32m    963[0m [38;5;28;01mwith[39;00m ev[38;5;241m.[39mtrigger_event([38;5;124m"[39m[38;5;124mnumba:compile[39m[38;5;124m"[39m, data[38;5;241m=[39mev_details):
[1;32m    964[0m     [38;5;28;01mtry[39;00m:
[0;32m--> 965[0m         cres [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_compiler[38;5;241m.[39mcompile(args, return_type)
[1;32m    966[0m     [38;5;28;01mexcept[39;00m errors[38;5;241m.[39mForceLiteralArg [38;5;28;01mas[39;00m e:
[1;32m    967[0m         [38;5;28;01mdef[39;00m [38;5;21mfolded[39m(args, kws):

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/dispatcher.py:129[0m, in [0;36m_FunctionCompiler.compile[0;34m(self, args, return_type)[0m
[1;32m    127[0m     [38;5;28;01mreturn[39;00m retval
[1;32m    128[0m [38;5;28;01melse[39;00m:
[0;32m--> 129[0m     [38;5;28;01mraise[39;00m retval

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/dispatcher.py:139[0m, in [0;36m_FunctionCompiler._compile_cached[0;34m(self, args, return_type)[0m
[1;32m    136[0m     [38;5;28;01mpass[39;00m
[1;32m    138[0m [38;5;28;01mtry[39;00m:
[0;32m--> 139[0m     retval [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_compile_core(args, return_type)
[1;32m    140[0m [38;5;28;01mexcept[39;00m errors[38;5;241m.[39mTypingError [38;5;28;01mas[39;00m e:
[1;32m    141[0m     [38;5;28mself[39m[38;5;241m.[39m_failed_cache[key] [38;5;241m=[39m e

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/dispatcher.py:152[0m, in [0;36m_FunctionCompiler._compile_core[0;34m(self, args, return_type)[0m
[1;32m    149[0m flags [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_customize_flags(flags)
[1;32m    151[0m impl [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_get_implementation(args, {})
[0;32m--> 152[0m cres [38;5;241m=[39m compiler[38;5;241m.[39mcompile_extra([38;5;28mself[39m[38;5;241m.[39mtargetdescr[38;5;241m.[39mtyping_context,
[1;32m    153[0m                               [38;5;28mself[39m[38;5;241m.[39mtargetdescr[38;5;241m.[39mtarget_context,
[1;32m    154[0m                               impl,
[1;32m    155[0m                               args[38;5;241m=[39margs, return_type[38;5;241m=[39mreturn_type,
[1;32m    156[0m                               flags[38;5;241m=[39mflags, [38;5;28mlocals[39m[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39mlocals,
[1;32m    157[0m                               pipeline_class[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39mpipeline_class)
[1;32m    158[0m [38;5;66;03m# Check typing error if object mode is used[39;00m
[1;32m    159[0m [38;5;28;01mif[39;00m cres[38;5;241m.[39mtyping_error [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m [38;5;129;01mand[39;00m [38;5;129;01mnot[39;00m flags[38;5;241m.[39menable_pyobject:

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler.py:762[0m, in [0;36mcompile_extra[0;34m(typingctx, targetctx, func, args, return_type, flags, locals, library, pipeline_class)[0m
[1;32m    738[0m [38;5;250m[39m[38;5;124;03m"""Compiler entry point[39;00m
[1;32m    739[0m 
[1;32m    740[0m [38;5;124;03mParameter[39;00m
[0;32m   (...)[0m
[1;32m    758[0m [38;5;124;03m    compiler pipeline[39;00m
[1;32m    759[0m [38;5;124;03m"""[39;00m
[1;32m    760[0m pipeline [38;5;241m=[39m pipeline_class(typingctx, targetctx, library,
[1;32m    761[0m                           args, return_type, flags, [38;5;28mlocals[39m)
[0;32m--> 762[0m [38;5;28;01mreturn[39;00m pipeline[38;5;241m.[39mcompile_extra(func)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler.py:460[0m, in [0;36mCompilerBase.compile_extra[0;34m(self, func)[0m
[1;32m    458[0m [38;5;28mself[39m[38;5;241m.[39mstate[38;5;241m.[39mlifted [38;5;241m=[39m ()
[1;32m    459[0m [38;5;28mself[39m[38;5;241m.[39mstate[38;5;241m.[39mlifted_from [38;5;241m=[39m [38;5;28;01mNone[39;00m
[0;32m--> 460[0m [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_compile_bytecode()

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler.py:528[0m, in [0;36mCompilerBase._compile_bytecode[0;34m(self)[0m
[1;32m    524[0m [38;5;250m[39m[38;5;124;03m"""[39;00m
[1;32m    525[0m [38;5;124;03mPopulate and run pipeline for bytecode input[39;00m
[1;32m    526[0m [38;5;124;03m"""[39;00m
[1;32m    527[0m [38;5;28;01massert[39;00m [38;5;28mself[39m[38;5;241m.[39mstate[38;5;241m.[39mfunc_ir [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m
[0;32m--> 528[0m [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_compile_core()

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler.py:507[0m, in [0;36mCompilerBase._compile_core[0;34m(self)[0m
[1;32m    505[0m         [38;5;28mself[39m[38;5;241m.[39mstate[38;5;241m.[39mstatus[38;5;241m.[39mfail_reason [38;5;241m=[39m e
[1;32m    506[0m         [38;5;28;01mif[39;00m is_final_pipeline:
[0;32m--> 507[0m             [38;5;28;01mraise[39;00m e
[1;32m    508[0m [38;5;28;01melse[39;00m:
[1;32m    509[0m     [38;5;28;01mraise[39;00m CompilerError([38;5;124m"[39m[38;5;124mAll available pipelines exhausted[39m[38;5;124m"[39m)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler.py:494[0m, in [0;36mCompilerBase._compile_core[0;34m(self)[0m
[1;32m    492[0m res [38;5;241m=[39m [38;5;28;01mNone[39;00m
[1;32m    493[0m [38;5;28;01mtry[39;00m:
[0;32m--> 494[0m     pm[38;5;241m.[39mrun([38;5;28mself[39m[38;5;241m.[39mstate)
[1;32m    495[0m     [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mstate[38;5;241m.[39mcr [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:
[1;32m    496[0m         [38;5;28;01mbreak[39;00m

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler_machinery.py:368[0m, in [0;36mPassManager.run[0;34m(self, state)[0m
[1;32m    365[0m msg [38;5;241m=[39m [38;5;124m"[39m[38;5;124mFailed in [39m[38;5;132;01m%s[39;00m[38;5;124m mode pipeline (step: [39m[38;5;132;01m%s[39;00m[38;5;124m)[39m[38;5;124m"[39m [38;5;241m%[39m \
[1;32m    366[0m     ([38;5;28mself[39m[38;5;241m.[39mpipeline_name, pass_desc)
[1;32m    367[0m patched_exception [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_patch_error(msg, e)
[0;32m--> 368[0m [38;5;28;01mraise[39;00m patched_exception

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler_machinery.py:356[0m, in [0;36mPassManager.run[0;34m(self, state)[0m
[1;32m    354[0m pass_inst [38;5;241m=[39m _pass_registry[38;5;241m.[39mget(pss)[38;5;241m.[39mpass_inst
[1;32m    355[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(pass_inst, CompilerPass):
[0;32m--> 356[0m     [38;5;28mself[39m[38;5;241m.[39m_runPass(idx, pass_inst, state)
[1;32m    357[0m [38;5;28;01melse[39;00m:
[1;32m    358[0m     [38;5;28;01mraise[39;00m [38;5;167;01mBaseException[39;00m([38;5;124m"[39m[38;5;124mLegacy pass in use[39m[38;5;124m"[39m)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler_lock.py:35[0m, in [0;36m_CompilerLock.__call__.<locals>._acquire_compile_lock[0;34m(*args, **kwargs)[0m
[1;32m     32[0m [38;5;129m@functools[39m[38;5;241m.[39mwraps(func)
[1;32m     33[0m [38;5;28;01mdef[39;00m [38;5;21m_acquire_compile_lock[39m([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[1;32m     34[0m     [38;5;28;01mwith[39;00m [38;5;28mself[39m:
[0;32m---> 35[0m         [38;5;28;01mreturn[39;00m func([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler_machinery.py:311[0m, in [0;36mPassManager._runPass[0;34m(self, index, pss, internal_state)[0m
[1;32m    309[0m     mutated [38;5;241m|[39m[38;5;241m=[39m check(pss[38;5;241m.[39mrun_initialization, internal_state)
[1;32m    310[0m [38;5;28;01mwith[39;00m SimpleTimer() [38;5;28;01mas[39;00m pass_time:
[0;32m--> 311[0m     mutated [38;5;241m|[39m[38;5;241m=[39m check(pss[38;5;241m.[39mrun_pass, internal_state)
[1;32m    312[0m [38;5;28;01mwith[39;00m SimpleTimer() [38;5;28;01mas[39;00m finalize_time:
[1;32m    313[0m     mutated [38;5;241m|[39m[38;5;241m=[39m check(pss[38;5;241m.[39mrun_finalizer, internal_state)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/compiler_machinery.py:273[0m, in [0;36mPassManager._runPass.<locals>.check[0;34m(func, compiler_state)[0m
[1;32m    272[0m [38;5;28;01mdef[39;00m [38;5;21mcheck[39m(func, compiler_state):
[0;32m--> 273[0m     mangled [38;5;241m=[39m func(compiler_state)
[1;32m    274[0m     [38;5;28;01mif[39;00m mangled [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m ([38;5;28;01mTrue[39;00m, [38;5;28;01mFalse[39;00m):
[1;32m    275[0m         msg [38;5;241m=[39m ([38;5;124m"[39m[38;5;124mCompilerPass implementations should return True/False. [39m[38;5;124m"[39m
[1;32m    276[0m                [38;5;124m"[39m[38;5;124mCompilerPass with name [39m[38;5;124m'[39m[38;5;132;01m%s[39;00m[38;5;124m'[39m[38;5;124m did not.[39m[38;5;124m"[39m)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typed_passes.py:110[0m, in [0;36mBaseTypeInference.run_pass[0;34m(self, state)[0m
[1;32m    104[0m [38;5;250m[39m[38;5;124;03m"""[39;00m
[1;32m    105[0m [38;5;124;03mType inference and legalization[39;00m
[1;32m    106[0m [38;5;124;03m"""[39;00m
[1;32m    107[0m [38;5;28;01mwith[39;00m fallback_context(state, [38;5;124m'[39m[38;5;124mFunction [39m[38;5;124m"[39m[38;5;132;01m%s[39;00m[38;5;124m"[39m[38;5;124m failed type inference[39m[38;5;124m'[39m
[1;32m    108[0m                       [38;5;241m%[39m (state[38;5;241m.[39mfunc_id[38;5;241m.[39mfunc_name,)):
[1;32m    109[0m     [38;5;66;03m# Type inference[39;00m
[0;32m--> 110[0m     typemap, return_type, calltypes, errs [38;5;241m=[39m type_inference_stage(
[1;32m    111[0m         state[38;5;241m.[39mtypingctx,
[1;32m    112[0m         state[38;5;241m.[39mtargetctx,
[1;32m    113[0m         state[38;5;241m.[39mfunc_ir,
[1;32m    114[0m         state[38;5;241m.[39margs,
[1;32m    115[0m         state[38;5;241m.[39mreturn_type,
[1;32m    116[0m         state[38;5;241m.[39mlocals,
[1;32m    117[0m         raise_errors[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39m_raise_errors)
[1;32m    118[0m     state[38;5;241m.[39mtypemap [38;5;241m=[39m typemap
[1;32m    119[0m     [38;5;66;03m# save errors in case of partial typing[39;00m

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typed_passes.py:86[0m, in [0;36mtype_inference_stage[0;34m(typingctx, targetctx, interp, args, return_type, locals, raise_errors)[0m
[1;32m     83[0m [38;5;28;01mfor[39;00m k, v [38;5;129;01min[39;00m [38;5;28mlocals[39m[38;5;241m.[39mitems():
[1;32m     84[0m     infer[38;5;241m.[39mseed_type(k, v)
[0;32m---> 86[0m infer[38;5;241m.[39mbuild_constraint()
[1;32m     87[0m [38;5;66;03m# return errors in case of partial typing[39;00m
[1;32m     88[0m errs [38;5;241m=[39m infer[38;5;241m.[39mpropagate(raise_errors[38;5;241m=[39mraise_errors)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typeinfer.py:1039[0m, in [0;36mTypeInferer.build_constraint[0;34m(self)[0m
[1;32m   1037[0m [38;5;28;01mfor[39;00m blk [38;5;129;01min[39;00m [38;5;28mself[39m[38;5;241m.[39mblocks[38;5;241m.[39mvalues():
[1;32m   1038[0m     [38;5;28;01mfor[39;00m inst [38;5;129;01min[39;00m blk[38;5;241m.[39mbody:
[0;32m-> 1039[0m         [38;5;28mself[39m[38;5;241m.[39mconstrain_statement(inst)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typeinfer.py:1386[0m, in [0;36mTypeInferer.constrain_statement[0;34m(self, inst)[0m
[1;32m   1384[0m [38;5;28;01mdef[39;00m [38;5;21mconstrain_statement[39m([38;5;28mself[39m, inst):
[1;32m   1385[0m     [38;5;28;01mif[39;00m [38;5;28misinstance[39m(inst, ir[38;5;241m.[39mAssign):
[0;32m-> 1386[0m         [38;5;28mself[39m[38;5;241m.[39mtypeof_assign(inst)
[1;32m   1387[0m     [38;5;28;01melif[39;00m [38;5;28misinstance[39m(inst, ir[38;5;241m.[39mSetItem):
[1;32m   1388[0m         [38;5;28mself[39m[38;5;241m.[39mtypeof_setitem(inst)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typeinfer.py:1461[0m, in [0;36mTypeInferer.typeof_assign[0;34m(self, inst)[0m
[1;32m   1458[0m     [38;5;28mself[39m[38;5;241m.[39mconstraints[38;5;241m.[39mappend(Propagate(dst[38;5;241m=[39minst[38;5;241m.[39mtarget[38;5;241m.[39mname,
[1;32m   1459[0m                                       src[38;5;241m=[39mvalue[38;5;241m.[39mname, loc[38;5;241m=[39minst[38;5;241m.[39mloc))
[1;32m   1460[0m [38;5;28;01melif[39;00m [38;5;28misinstance[39m(value, (ir[38;5;241m.[39mGlobal, ir[38;5;241m.[39mFreeVar)):
[0;32m-> 1461[0m     [38;5;28mself[39m[38;5;241m.[39mtypeof_global(inst, inst[38;5;241m.[39mtarget, value)
[1;32m   1462[0m [38;5;28;01melif[39;00m [38;5;28misinstance[39m(value, ir[38;5;241m.[39mArg):
[1;32m   1463[0m     [38;5;28mself[39m[38;5;241m.[39mtypeof_arg(inst, inst[38;5;241m.[39mtarget, value)

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typeinfer.py:1561[0m, in [0;36mTypeInferer.typeof_global[0;34m(self, inst, target, gvar)[0m
[1;32m   1559[0m [38;5;28;01mdef[39;00m [38;5;21mtypeof_global[39m([38;5;28mself[39m, inst, target, gvar):
[1;32m   1560[0m     [38;5;28;01mtry[39;00m:
[0;32m-> 1561[0m         typ [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mresolve_value_type(inst, gvar[38;5;241m.[39mvalue)
[1;32m   1562[0m     [38;5;28;01mexcept[39;00m TypingError [38;5;28;01mas[39;00m e:
[1;32m   1563[0m         [38;5;28;01mif[39;00m (gvar[38;5;241m.[39mname [38;5;241m==[39m [38;5;28mself[39m[38;5;241m.[39mfunc_id[38;5;241m.[39mfunc_name
[1;32m   1564[0m                 [38;5;129;01mand[39;00m gvar[38;5;241m.[39mname [38;5;129;01min[39;00m _temporary_dispatcher_map):
[1;32m   1565[0m             [38;5;66;03m# Self-recursion case where the dispatcher is not (yet?) known[39;00m
[1;32m   1566[0m             [38;5;66;03m# as a global variable[39;00m

File [0;32m~/anaconda3/lib/python3.11/site-packages/numba/core/typeinfer.py:1482[0m, in [0;36mTypeInferer.resolve_value_type[0;34m(self, inst, val)[0m
[1;32m   1480[0m [38;5;28;01mexcept[39;00m [38;5;167;01mValueError[39;00m [38;5;28;01mas[39;00m e:
[1;32m   1481[0m     msg [38;5;241m=[39m [38;5;28mstr[39m(e)
[0;32m-> 1482[0m [38;5;28;01mraise[39;00m TypingError(msg, loc[38;5;241m=[39minst[38;5;241m.[39mloc)

[0;31mTypingError[0m: Failed in nopython mode pipeline (step: nopython frontend)
[1mNameError: name 'etree' is not defined[0m
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
[1mNameError: name 'etree' is not defined[0m

